Documentation/ApiOverview/Events/Events/Backend/_BeforeRecordDownloadIsExecutedEvent/_DataListener.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyVendor\MyExtension\Backend\EventListener;
+
+use TYPO3\CMS\Backend\RecordList\Event\BeforeRecordDownloadIsExecutedEvent;
+use TYPO3\CMS\Core\Attribute\AsEventListener;
+
+#[AsEventListener(identifier: 'my-package/record-list-download-data')]
+final readonly class DataListener
+{
+    public function __invoke(BeforeRecordDownloadIsExecutedEvent $event): void
+    {
+        // List of redactable fields.
+        $gdprFields = ['title', 'author'];
+
+        $headerRow = $event->getHeaderRow();
+        $records = $event->getRecords();
+
+        // Iterate header to mark redacted fields...
+        foreach ($headerRow as $headerRowKey => $headerRowValue) {
+            if (in_array($headerRowKey, $gdprFields, true)) {
+                $headerRow[$headerRowKey] .= ' (REDACTED)';
+            }
+        }
+
+        // Redact actual content...
+        foreach ($records as $uid => $record) {
+            foreach ($gdprFields as $gdprField) {
+                if (isset($record[$gdprField])) {
+                    $records[$uid][$gdprField] = '(REDACTED)';
+                }
+            }
+        }
```

Comment:
```suggestion
        foreach ($records as $rowKey => $record) {
            foreach ($gdprFields as $gdprField) {
                if (isset($record[$gdprField])) {
                    $records[$rowKey][$gdprField] = '(REDACTED)';
                }
            }
        }
```

Reply:
In Core each line will be added via `$records[]`, so there is no record UID available

Reply:
Done, thanks for catching it